core/src/main/java/org/apache/carbondata/core/indexstore/ExtendedBlocklet.java

```diff
@@ -221,7 +223,13 @@ public void deserializeFields(DataInput in, String[] locations, String tablePath
       indexUniqueId = in.readUTF();
     }
     String filePath = getPath();
-    if (filePath.startsWith(File.separator)) {
+    boolean isLocalFile = FileFactory.getCarbonFile(filePath) instanceof LocalCarbonFile;
+    // If it is external segment path, table path need not be appended to filePath
+    // Example filepath: hdfs://hacluster/opt/newsegmentpath/
+    // filePath value would start with hdfs:// or s3:// . If it is local
+    // ubuntu storage, it starts with File separator, so check if given path exists or not.
+    if ((!isLocalFile && filePath.startsWith(File.separator)) || (isLocalFile && !FileFactory
```

Comment:
the comment is not clear, please rewrite it with better example and scenarios

Reply:
ok done